### PR TITLE
filter keypoint correspondences as well as R,t in cycle consistency

### PR DESCRIPTION
**Pull request by johnwlambert**
Filter out bad keypoint correspondences/matches between (i1,i2) when filtering out bad R,t also.

### CODE REVIEW

**Comment by johnwlambert** [comment]
Appears to lead to a significant improvement @akshay-krishnan @ayushbaid:

on skydio-32 with deep front end
| metric name | on master | on this branch |
| :---| :--: | :--:  |
| number_tracks_unfiltered |	5182  | 5246 |
| mean_3d_track_lengths_unfiltered	| 2.65901 | 2.69272 |
| mean_reprojection_errors_unfiltered_px	| 10.477 | 7.78021 |
| mean_rotation_error_angle_deg |	7.49793 | 7.99806 |
| mean_translation_error_distance |	0.565324 | 0.472336 |
| mean_translation_angle_error_deg |	10.3642 | 8.26896 |
| number_tracks_filtered |	1354 | 2241 |
| mean_3d_track_lengths_filtered |	2.07016 | 2.10174 |
| mean_reprojection_errors_filtered_px |	1.35429 | 1.29525 |

reproj error unfilftered 10.5 -> 7.8
note trans error dist 0.57 -> 0.47
trans angle err 10.4 -> 8.3

rotation error increases slightly, 7.5 -> 7.999

File: gtsfm/scene_optimizer.py [hunk 1]
```diff
@@ -177,10 +177,11 @@ def create_computation_graph(
 
         # ensure cycle consistency in triplets
         cycle_consistent_graph = dask.delayed(cycle_consistency.filter_to_cycle_consistent_edges)(
-            i2Ri1_graph_dict, i2Ui1_graph_dict, two_view_reports_dict
+            i2Ri1_graph_dict, i2Ui1_graph_dict, v_corr_idxs_graph_dict, two_view_reports_dict
         )
         i2Ri1_graph_dict = cycle_consistent_graph[0]
         i2Ui1_graph_dict = cycle_consistent_graph[1]
+        v_corr_idxs_graph_dict = cycle_consistent_graph[2]
```
Comment (akshay-krishnan) [inline gtsfm/scene_optimizer.py:184]: you can avoid this and actually unpack the from the delayed object by specifying the nout= parameter.  Like: 
a, b, c = dask.delayed(myfunc, nout=3)(x, y)

File: gtsfm/scene_optimizer.py [hunk 2]
```diff
@@ -177,10 +177,11 @@ def create_computation_graph(
...
```
Comment (akshay-krishnan) [inline gtsfm/scene_optimizer.py:185]: curious why we have cycle consistency check in scene optimizer? Seems like it should belong in MVO?

**Review by akshay-krishnan — COMMENTED** [review]
PR description seems to indicate R, t but I dont see filtering `t` ...

File: gtsfm/scene_optimizer.py (hunk 2 above)
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py:185]: could go in either, i don't have too strong of feelings about where it goes

File: gtsfm/scene_optimizer.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py]: @akshay-krishnan good point, should be cleaned up now

**Comment by johnwlambert** [comment]
@ayushbaid @akshay-krishnan gentle reminder about reviewing this one : - )

File: gtsfm/scene_optimizer.py (hunk 2 above)
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py:182]: Created an issue here: https://github.com/borglab/gtsfm/issues/289

**Review by ayushbaid — APPROVED** [review]

**Comment by johnwlambert** [comment]
Thanks @ayushbaid !

**Pull request merged**